projects/bp-gallery/src/components/views/location-curating/LocationPanel.tsx

```diff
@@ -1,39 +1,60 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useSimplifiedQueryResponseData } from '../../../graphql/queryUtils';
+import { useVisit } from '../../../helpers/history';
 import useGenericTagEndpoints from '../../../hooks/generic-endpoints.hook';
 import { FlatTag, TagType } from '../../../types/additionalFlatTypes';
 import Loading from '../../common/Loading';
 import ProtectedRoute from '../../common/ProtectedRoute';
 import QueryErrorDisplay from '../../common/QueryErrorDisplay';
 import AddLocationEntry from './AddLocationEntry';
+import { useFoldoutStatus } from './FoldoutStatusContext';
 import LocationBranch from './LocationBranch';
 import LocationPanelHeader from './LocationPanelHeader';
 import { LocationPanelPermissionsProvider } from './LocationPanelPermissionsProvider';
 import { useCreateNewTag } from './location-management-helpers';
 import { useGetTagStructures } from './tag-structure-helpers';
 
 const setUnacceptedSubtagsCount = (tag: FlatTag) => {
-  if (!tag.child_tags?.length) {
-    return tag.accepted ? 0 : 1;
-  }
   let subtagCount = 0;
-  tag.child_tags.forEach((childTag: FlatTag) => {
+  tag.child_tags?.forEach((childTag: FlatTag) => {
     subtagCount += setUnacceptedSubtagsCount(childTag);
   });
   tag.unacceptedSubtags = subtagCount;
-  return subtagCount;
+  return (tag.accepted ? 0 : 1) + subtagCount;
 };
 
 const LocationPanel = () => {
   const { t } = useTranslation();
+  const { location } = useVisit();
+  const foldoutStatus = useFoldoutStatus();
 
   const { allTagsQuery, canUseTagTableViewQuery } = useGenericTagEndpoints(TagType.LOCATION);
 
   const { data, loading, error, refetch } = allTagsQuery();
   const flattened = useSimplifiedQueryResponseData(data);
   const flattenedTags: FlatTag[] | undefined = flattened ? Object.values(flattened)[0] : undefined;
 
+  useEffect(() => {
+    if (!foldoutStatus) {
+      return;
+    }
+    foldoutStatus.current = Object.fromEntries(
+      flattenedTags
+        ? flattenedTags.map(tag => [
+            tag.id,
+            {
+              isOpen:
+                location.state?.openBranches && tag.id in location.state.openBranches
+                  ? location.state.openBranches[tag.id].isOpen
+                  : false,
+            },
+          ])
+        : []
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Maybe add a comment here why you don't want dependencies? I guess you only want to trigger this effect at first render?